core/node/libp2p/rcmgr_defaults.go

```diff
@@ -4,67 +4,37 @@ import (
 	"fmt"
 
 	"github.com/dustin/go-humanize"
+	"github.com/ipfs/kubo/config"
+	"github.com/ipfs/kubo/core/node/libp2p/fd"
 	"github.com/libp2p/go-libp2p"
 	rcmgr "github.com/libp2p/go-libp2p/p2p/host/resource-manager"
 	"github.com/pbnjay/memory"
-
-	"github.com/ipfs/kubo/config"
-	"github.com/ipfs/kubo/core/node/libp2p/fd"
 )
 
-// We are doing some magic when parsing config files (we are using a map[string]interface{} to compare config files).
-// When you don't have a type the JSON Parse function cast numbers to float64 by default,
-// losing precision when writing the final number. So if we use math.MaxInt as our infinite number,
-// after writing the config file we will have 9223372036854776000 instead of 9223372036854775807,
-// making the parsing process fail. Setting 1e9 (1000000000) as "no limit" value. It also avoids to overflow on 32 bit architectures.
-const bigEnough = 1e9
-
-var infiniteBaseLimit = rcmgr.BaseLimit{
-	Streams:         bigEnough,
-	StreamsInbound:  bigEnough,
-	StreamsOutbound: bigEnough,
-	Conns:           bigEnough,
-	ConnsInbound:    bigEnough,
-	ConnsOutbound:   bigEnough,
-	FD:              bigEnough,
-	Memory:          bigEnough,
-}
-
-var noLimitIncrease = rcmgr.BaseLimitIncrease{
-	ConnsInbound:    0,
-	ConnsOutbound:   0,
-	Conns:           0,
-	StreamsInbound:  0,
-	StreamsOutbound: 0,
-	Streams:         0,
-	Memory:          0,
-	FDFraction:      0,
-}
+var infiniteResrouceLimits = rcmgr.InfiniteLimits.ToPartialLimitConfig().System
 
 // This file defines implicit limit defaults used when Swarm.ResourceMgr.Enabled
 
 // createDefaultLimitConfig creates LimitConfig to pass to libp2p's resource manager.
 // The defaults follow the documentation in docs/libp2p-resource-management.md.
 // Any changes in the logic here should be reflected there.
-func createDefaultLimitConfig(cfg config.SwarmConfig) (rcmgr.ConcreteLimitConfig, error) {
+func createDefaultLimitConfig(cfg config.SwarmConfig) (limits rcmgr.ConcreteLimitConfig, logMessageForStartup string, err error) {
 	maxMemoryDefaultString := humanize.Bytes(uint64(memory.TotalMemory()) / 2)
 	maxMemoryString := cfg.ResourceMgr.MaxMemory.WithDefault(maxMemoryDefaultString)
 	maxMemory, err := humanize.ParseBytes(maxMemoryString)
 	if err != nil {
-		return rcmgr.ConcreteLimitConfig{}, err
+		return rcmgr.ConcreteLimitConfig{}, "", err
 	}
 
 	maxMemoryMB := maxMemory / (1024 * 1024)
 	maxFD := int(cfg.ResourceMgr.MaxFileDescriptors.WithDefault(int64(fd.GetNumFDs()) / 2))
 
-	// We want to see this message on startup, that's why we are using fmt instead of log.
-	fmt.Printf(`
+	msg := fmt.Sprintf(`
 Computing default go-libp2p Resource Manager limits based on:
```

Comment:
```suggestion
Computed default go-libp2p Resource Manager limits based on:
```

Changed to past tense since by the time this message is printed, the computing will have already happened.

What about also moving this message definition down to the bottom of this function closer to the return statement?

it still feels weird too to be generating this message here.  I think ideally we'd return the maxMemory and maxFD that were used to do the computation and let the caller generate the message.  Your call on how clean you want to make this.